packages/pages/src/components/analytics/hooks.ts

```diff
@@ -19,12 +19,11 @@ declare global {
  *
  * @public
  */
-export function useAnalytics(): AnalyticsMethods {
+export function useAnalytics(): AnalyticsMethods | null {
   const ctx = useContext(AnalyticsContext);
+
   if (!ctx) {
-    throw new Error(
```

Comment:
Why remove the error and return null? Seems a little weird you need to null check everywhere now.

Reply:
So in adding analytics to the Link component we realized this was an anti-pattern because it means that any component where we to automatically add analytics (e.g. the Link component) would only ever be able to be called as a child of an `AnalyticsProvider` or it would throw an error.  You can't call a hook inside of a conditional so unless we want to make the Link component useless outside of the context of an AnalyticsProvider we have to do it this way.  

I think in the future we could refactor the Analytics class so that it doesn't keep track of it's own state and instead move the state into the useAnalytics hook so that useAnalytics never returns null, but you'd still have the problem of the track method and other methods not doing anything unless you've given the template data to Analytics at some point prior to calling the `track` method.